### PR TITLE
port 2 requested things related to refernces

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request
port [this](https://github.com/tgstation/tgstation/pull/57728) and [that](https://github.com/tgstation/tgstation/pull/58972)

It compiles, but that about all I tested. 

## Why It's Good For The Game

it was requested

## Changelog
:cl: LemonInTheDark
refactor: 2 ports relating to refernce's 
/:cl:

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -76,6 +76,13 @@
 #define TESTING
 #endif
 
+//#if defined(UNIT_TESTS)
+//Hard del testing defines
+//#define REFERENCE_TRACKING
+//#define REFERENCE_TRACKING_DEBUG
+//#define FIND_REF_NO_CHECK_TICK
+//#endif
```
Comment (Archanial) [inline code/_compile_options.dm]: ?

Comment (ItsTisme) [reply code/_compile_options.dm]: These defines enable a proc that calls functions for the unit test, but said proc is not present in our code. I could try to port that wasent sure if we should or not

File: code/__DEFINES/_tick.dm [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+
+//!!!!!!!!!!!!!INCOMING CHANGE!!!!!!!!!!!!!!!!
+/// internal_tick_usage is updated every tick by extools
+#ifdef USE_EXTOOLS
+#define MAPTICK_LAST_INTERNAL_TICK_USAGE ((GLOB.internal_tick_usage / world.tick_lag) * 100)
+#else
+#define MAPTICK_LAST_INTERNAL_TICK_USAGE (world.map_cpu)
+#endif
+/// Tick limit while running normally
+//!!!!!!!!!!!!!INCOMING CHANGE END!!!!!!!!!!!!!!!
```
Comment (Archanial) [inline code/__DEFINES/_tick.dm]: we do be using auxtools you know

**Comment by ike709** [comment]
Fails CI

**Comment by ItsTisme** [comment]
> Fails CI

not sure why

**Comment by ItsTisme** [comment]
seems to work now

**Comment by Autisem** [comment]
seems to work now

File: code/datums/datum.dm [hunk 3]
```diff
@@ -42,9 +42,13 @@
 	var/list/cooldowns
 
 
-#ifdef TESTING
+#ifdef REFERENCE_TRACKING_DEBUG
 	var/running_find_references
 	var/last_find_references = 0
+	#ifdef REFERENCE_TRACKING_DEBUG
```
Comment (BrumalLake) [inline code/datums/datum.dm:48]: checking for the same thing as the outer if?

**Comment by ike709** [comment]
🔑 

This is a pretty sus port job

**Comment by Archanial** [comment]
sus port 🔑

**Pull request closed without merging**